src/util/queryString.ts

```diff
@@ -0,0 +1,47 @@
+import { SUPPORT_LANGUAGES } from '../constants';
+import { MAPPED_PLACES } from '../domain/eventSearch/constants';
+import { assertUnreachable } from './typescript.utils';
+
+// Add more known query parameters here
+export type QueryParam = 'places' | 'returnPath';
+
+export type QueryEntry = {
+  param: QueryParam;
+  value?: string;
+};
+
+const langPathRegExp = new RegExp(
+  `/(${Object.values(SUPPORT_LANGUAGES).join('|')})`
+);
+
+const stripLanguageFromPath = (path: string) =>
+  path.replace(langPathRegExp, '');
+
+const getParamValue = ({ param, value }: Required<QueryEntry>) => {
+  switch (param) {
+    case 'places':
+      return MAPPED_PLACES[value];
+    case 'returnPath':
+      return stripLanguageFromPath(value);
+    default:
+      return assertUnreachable(param, 'Unknown query parameter');
+  }
+};
+
+export const addEntriesToQueryString = (
+  queryString: string,
+  ...entries: Array<QueryEntry>
```

Comment:
The way these entries are passed bothers me little bit:

`addEntriesToQueryString('string', {param, value}, {param, value}, {param, value}, {param, value}, ...)`

Would it be better to just have single object with key-value pairs?

```
addEntriesToQueryString(
'string', 
{ places: 'param', returnPath: 'param }  
 ...
 )
```

Reply:
The idea was for the method to be common enough to accept any type of atributes, also queryParams can also have multiple parameters of the same type, like:
```  
addEntriesToQueryString(
    'string',
    { param: 'returnPath', value: 'events' },
    { param: 'returnPath', value: '/event/helsinki:af3pnza3zi' }
  );
  ```

Reply:
With multiple parameters of the same type I think array would do. But maybe my opinion comes from how some other query params builders work in other libraries

Reply:
Ah I thought I did not send the previous message when I started to investigate further. Now there is resolution with Object type and value can be either string or array of strings.